sqlmesh/integrations/dlt.py

```diff
@@ -85,6 +89,29 @@ def generate_dlt_models_and_settings(
     return sqlmesh_models, format_config(configs, db_type), get_start_date(storage_ids)
 
 
+def update_dlt_models(
+    context: Context, pipeline_name: str, update_tables: t.List[str], force: bool
+) -> str:
+    from sqlmesh.cli.example_project import _create_models
+
+    sqlmesh_models, _, _ = generate_dlt_models_and_settings(
+        pipeline_name=pipeline_name,
+        dialect=context.config.dialect or "",
+        update_tables=update_tables if update_tables else None,
+    )
+
+    if not update_tables and not force:
+        existing_models = [m.name for m in context.models.values()]
+        sqlmesh_models = {model for model in sqlmesh_models if model[0] not in existing_models}
+
+    if sqlmesh_models:
+        _create_models(models_path=context.path / "models", models=sqlmesh_models)
+        model_names = "\n".join([f"- {model[0]}" for model in sqlmesh_models])
+        return f"Updated SQLMesh models:\n{model_names}"
```

Comment:
I think the better API would be returning a set of model names here and then handling actual printing upstream. Alternatively, you can just print into console here directly and not return anything.

Reply:
Right I’ll refactor this to return the model names instead.